tests/scalers/gitlab_runner/gitlab_runner_test.go

```diff
@@ -0,0 +1,503 @@
+//go:build e2e
```

Comment:
Reading the e2e test, it waits a PAT so I guess that we need to create an e2e account to run it. Does it have any special requirement? Does it make sense to use gitlab.com instead of spin up a dummy gitlab instance in the cluster during the e2e test? https://docs.gitlab.com/charts/installation/deployment.html